### ISSUE TITLE
Discover step fails if beakerlib library not found in the repo

### ISSUE THREAD

**Issue by sopos** [psss/tmt#397]
If there's an existing repo on `github.com/beakerlib/<COMPONENT>` but the library is not there the discover step fails as git command cannot find the library there. This way it even does not try to satisfy the requirements from possibly enabled dnf repo.
I.e. if I want to move one of the distribution libraries to github I will break the others.
The correct behaviour should be to continue and try to satisfy the library requirement in different execution phases.

**Comment by psss** [comment]
This should be already fixed by #382. What version of `tmt` are you testing with?

**Comment by sopos** [comment]
```
$ rpm -q tmt
tmt-1.0-1.20201007135237127416.master.37.gb5844a7.fc31.noarch

$ tmt run -vvvddd -a provision -h connect -g 10.0.138.145 prepare -h shell -s 'dnf config-manager --enable beaker-tasks' execute -h beakerlib.tmt tests --name .
...
    discover
        workdir: /var/tmp/tmt/run-011/plans/default/discover
        how: fmf
        order: 50
            Run command 'git rev-parse --show-toplevel'.
            out: /home/dapospis/test/git/SP_components/tests
        directory: /home/dapospis/test/git/SP_components/tests
            Copy '/home/dapospis/test/git/SP_components/tests' to '/var/tmp/tmt/run-011/plans/default/discover/default/tests'.
            Check metadata tree in '/var/tmp/tmt/run-011/plans/default/discover/default/tests/'.
            Detected library 'library(distribution/dpcommon)'.
            Fetch library 'distribution/dpcommon'.
            Run command 'git clone https://github.com/beakerlib/distribution /var/tmp/tmt/run-011/plans/default/discover/default/libs/distribution'.
            err: Cloning into '/var/tmp/tmt/run-011/plans/default/discover/default/libs/distribution'...
            err: remote: Invalid username or password.
            err: fatal: Authentication failed for 'https://github.com/beakerlib/distribution/'
            Command returned '128'.
            Repository 'https://github.com/beakerlib/distribution' not found.
            Detected library 'library(epel/epel)'.
            Fetch library 'epel/epel'.
            Run command 'git clone https://github.com/beakerlib/epel /var/tmp/tmt/run-011/plans/default/discover/default/libs/epel'.
            err: Cloning into '/var/tmp/tmt/run-011/plans/default/discover/default/libs/epel'...
            Run command 'git checkout master'.
            err: error: pathspec 'master' did not match any file(s) known to git
            Command returned '1'.
```

**Comment by psss** [comment]
The problem above is that the epel repository does not have a `master` branch. It only has the `main` one, which seems to be the new github default for main branches. So far (according to the spec) we expect only master. Let's fix the repo. We could also check for `main` in the future but that needs some discussion.

**Comment by sopos** [comment]
I did not even notice that. Shouldn't tmt look at the default branch instead of hard-coded _master_?

**Comment by sopos** [comment]
ok, adding master branch moved me a bit forward
```
...
go
    discover
        workdir: /var/tmp/tmt/run-012/plans/default/discover
        how: fmf
        order: 50
            Run command 'git rev-parse --show-toplevel'.
            out: /home/dapospis/test/git/SP_components/tests
        directory: /home/dapospis/test/git/SP_components/tests
            Copy '/home/dapospis/test/git/SP_components/tests' to '/var/tmp/tmt/run-012/plans/default/discover/default/tests'.
            Check metadata tree in '/var/tmp/tmt/run-012/plans/default/discover/default/tests/'.
            Detected library 'library(distribution/dpcommon)'.
            Fetch library 'distribution/dpcommon'.
            Run command 'git clone https://github.com/beakerlib/distribution /var/tmp/tmt/run-012/plans/default/discover/default/libs/distribution'.
            err: Cloning into '/var/tmp/tmt/run-012/plans/default/discover/default/libs/distribution'...
            err: remote: Invalid username or password.
            err: fatal: Authentication failed for 'https://github.com/beakerlib/distribution/'
            Command returned '128'.
            Repository 'https://github.com/beakerlib/distribution' not found.
            Detected library 'library(epel/epel)'.
            Fetch library 'epel/epel'.
            Run command 'git clone https://github.com/beakerlib/epel /var/tmp/tmt/run-012/plans/default/discover/default/libs/epel'.
            err: Cloning into '/var/tmp/tmt/run-012/plans/default/discover/default/libs/epel'...
            Run command 'git checkout master'.
            err: Already on 'master'
            out: Your branch is up to date with 'origin/master'.
    finish
        workdir: /var/tmp/tmt/run-012/plans/default/finish
        summary: 0 tasks completed
        status: done
        Write file '/var/tmp/tmt/run-012/plans/default/finish/step.yaml'.
Traceback (most recent call last):
  File "/usr/lib/python3.7/site-packages/tmt/beakerlib.py", line 105, in fetch
    library = self.parent._library_cache[self.repo]
KeyError: 'epel'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/bin/tmt", line 8, in <module>
    tmt.cli.main()
  File "/usr/lib/python3.7/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/usr/lib/python3.7/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/usr/lib/python3.7/site-packages/click/core.py", line 1259, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/usr/lib/python3.7/site-packages/click/core.py", line 1290, in invoke
    return _process_result(rv)
  File "/usr/lib/python3.7/site-packages/click/core.py", line 1224, in _process_result
    value = ctx.invoke(self.result_callback, value, **ctx.params)
  File "/usr/lib/python3.7/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/usr/lib/python3.7/site-packages/click/decorators.py", line 21, in new_func
    return f(get_current_context(), *args, **kwargs)
  File "/usr/lib/python3.7/site-packages/tmt/cli.py", line 254, in finito
    context.obj.run.go()
  File "/usr/lib/python3.7/site-packages/tmt/base.py", line 944, in go
    plan.go()
  File "/usr/lib/python3.7/site-packages/tmt/base.py", line 536, in go
    step.go()
  File "/usr/lib/python3.7/site-packages/tmt/steps/discover/__init__.py", line 124, in go
    plugin.go()
  File "/usr/lib/python3.7/site-packages/tmt/steps/discover/fmf.py", line 165, in go
    test.require, test.recommend, parent=self)
  File "/usr/lib/python3.7/site-packages/tmt/beakerlib.py", line 180, in dependencies
    library = Library(dependency, parent=parent)
  File "/usr/lib/python3.7/site-packages/tmt/beakerlib.py", line 91, in __init__
    self.fetch()
  File "/usr/lib/python3.7/site-packages/tmt/beakerlib.py", line 140, in fetch
    self.tree = fmf.Tree(directory)
  File "/usr/lib/python3.7/site-packages/fmf/base.py", line 75, in __init__
    self._initialize(path=data)
  File "/usr/lib/python3.7/site-packages/fmf/base.py", line 101, in _initialize 
    os.path.abspath(path)))
fmf.utils.RootError: Unable to find tree root for '/var/tmp/tmt/run-012/plans/default/discover/default/libs/epel'.
```

**Comment by sopos** [comment]
just for the record, previously I was using `tmt-0.21-1.fc31.noarch`

**Comment by psss** [comment]
Ok, we should nicely handle when there are no fmf metadata in the library repository. Should be fixed by #399.
